### PR TITLE
mongodb: fix processing datetime object

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24276

## Description
MongoDB returns datetime objects from the DB. so from now on, they will be converted to strings. Same as input, just put the date string and it'll automatically convert it to DateTime.

Added a step to both unitest and test playbook.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py [hunk 1]
```diff
@@ -73,37 +78,70 @@ def normalize_id(entry: dict):
         Examples:
             >>> Client.normalize_id({'_id': ObjectId('5e4412f230c5b8f63a7356ba')})
             {'_id': '5e4412f230c5b8f63a7356ba'}
-        '''
+        """
         entry['_id'] = str(entry.pop('_id'))
         return entry
 
+    @classmethod
+    def datetime_to_str(cls, obj):
```
Comment (Itay4) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB.py:86]: add type hints

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py [hunk 2]
```diff
@@ -73,37 +78,70 @@ def normalize_id(entry: dict):
...
+        """ Converts any object with date value of type datetime to str
+
+        Args:
+            obj: The object contains possible date values.
+
+        Returns:
+            All Dates from type datetime converted to str in this format - %Y-%m-%dT%H:%M:%S.000Z\
```
Comment (Itay4) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB.py:93]:
```suggestion
            All Dates from type datetime converted to str in this format - %Y-%m-%dT%H:%M:%S.000Z
```

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py [hunk 3]
```diff
@@ -231,21 +291,34 @@ def insert_entry_command(
         raise DemistoException('The `entry` argument is not a valid json.')
 
 
+def validate_json_objects(json):
```
Comment (Itay4) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB.py:294]: add type hints

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py [hunk 4]
```diff
@@ -129,6 +167,27 @@ def _convert(entry: dict):
     return _convert(entries)
 
 
+def convert_str_to_datetime(entries):
```
Comment (Itay4) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB.py:170]: add type hints

File: Packs/MongoDB/Integrations/MongoDB/MongoDB_test.py [hunk 5]
```diff
@@ -33,3 +36,74 @@ def test_convert_id_to_object_id(func_input, expected):
 @pytest.mark.parametrize("func_input, expected", object_to_id)
 def test_convert_object_id_to_str(func_input, expected):
     assert expected == convert_object_id_to_str(func_input)
+
+
```
Comment (Itay4) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB_test.py:40]: for all tests add docstring in template of `Given-When-Then`

File: Packs/MongoDB/Integrations/MongoDB/MongoDB_test.py [hunk 6]
```diff
@@ -33,3 +36,74 @@ def test_convert_id_to_object_id(func_input, expected):
...
+        res = convert_str_to_datetime(func_input)
+        assert isinstance(res['time'], datetime)
+
+    def test_convert_str_to_datetime_no_datetime_obj(self):
+        inputs = {1: 2}
+        res = convert_str_to_datetime(inputs)
+        assert isinstance(res[1], int)
+
+
+class TestDatetimeToStr:
```
Comment (Itay4) [inline Packs/MongoDB/Integrations/MongoDB/MongoDB_test.py:65]: add test for object of type `Cursor`

File: Packs/MongoDB/pack_metadata.json [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
-  ],
-  "tags": [],
-  "useCases": [],
-  "keywords": [
-    "MongoDB"
-  ]
+    "name": "MongoDB",
+    "description": "Use the MongoDB integration to search and query entries in your MongoDB.",
+    "support": "xsoar",
+    "currentVersion": "1.1.0",
```
Comment (Itay4) [inline Packs/MongoDB/pack_metadata.json:5]: this should be a revision, change to `1.0.1`

File: Packs/MongoDB/Integrations/MongoDB/MongoDB.py (hunk 1 above)
Comment (jochman) [reply Packs/MongoDB/Integrations/MongoDB/MongoDB.py:86]: It'll get any and will return any... but fine

File: Packs/MongoDB/Integrations/MongoDB/MongoDB_test.py (hunk 6 above)
Comment (jochman) [reply Packs/MongoDB/Integrations/MongoDB/MongoDB_test.py:65]: There's a problem creating this object outside mongodb. it anyway reacts like a list.

**Comment by jochman** [comment]
https://github.com/demisto/content/pull/7138

**Pull request closed without merging**